tmt/steps/report/junit.py

```diff
@@ -58,40 +67,230 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
+        if flavor != CUSTOM_FLAVOR_NAME:
+            xsd_schema_path = Path(tmt.utils.resource_files(
+                Path(f'steps/report/junit/schemas/{flavor}.xsd')))
+
+            schema_root: XMLElement = etree.XML(xsd_schema_path.read_bytes())
+            xml_parser_kwargs['schema'] = etree.XMLSchema(schema_root)
+        else:
+            phase.warn(
+                f"The '{CUSTOM_FLAVOR_NAME}' JUnit flavor is used, you are solely responsible "
+                "for the validity of the XML schema.")
+
+            phase.warn(f"The pretty print is always disabled for '{CUSTOM_FLAVOR_NAME}' JUnit "
+                       "flavor")
+            xml_parser_kwargs['remove_blank_text'] = prettify = False
+
+        xml_parser = etree.XMLParser(**xml_parser_kwargs)
 
-    for result in report.step.plan.execute.results():
         try:
-            main_log = report.step.plan.execute.read(result.log[0])
-        except (IndexError, AttributeError):
-            main_log = None
-        case = junit_xml.TestCase(
-            result.name,
-            classname=None,
-            elapsed_sec=duration_to_seconds(result.duration))
-
-        if report.data.include_output_log:
-            case.stdout = main_log
-
-        # Map tmt OUTCOME to JUnit states
-        if result.result == tmt.result.ResultOutcome.ERROR:
-            case.add_error_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.FAIL:
-            case.add_failure_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.INFO:
-            case.add_skipped_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.WARN:
-            case.add_error_info(result.result.value, output=result.failures(main_log))
-        # Passed state is the default
-        suite.test_cases.append(case)
-
-    return suite
+            # S320: Parsing of untrusted data is known to be vulnerable to XML
+            # attacks.
+            tree_root: XMLElement = etree.fromstring(xml_data, xml_parser)  # noqa: S320
+
+        except etree.XMLSyntaxError as e:
+            phase.warn(
+                'The generated XML output is not a valid XML file or it is not valid against the '
+                'XSD schema.')
+
+            if flavor != CUSTOM_FLAVOR_NAME:
+                phase.warn('Please, report this problem to project maintainers.')
+
+            for err in e.error_log:
+                phase.warn(str(err))
+
+            # Return the prettified XML without checking the XSD
+            del xml_parser_kwargs['schema']
+
+            try:
+                tree_root = etree.fromstring(  # noqa: S320
+                    xml_data, etree.XMLParser(**xml_parser_kwargs))
+            except etree.XMLSyntaxError as e:
+                phase.warn(str(e))
+                phase.verbose('data', xml_data, 'red')
```

Comment:
`data` as a label is often pretty useless because it's so overused. Would there be a different label we could use, maybe `rendered XML` or something along that line?